### PR TITLE
Section 3.4 - 3.7 Completed

### PR DESCRIPTION
**Pull request by PKALXI**
Completed 3.4 - 3.7

### CODE REVIEW

File: docs/VnVPlan/VnVPlan.tex [hunk 1]
```diff
@@ -147,6 +151,16 @@ \subsection{Verification and Validation Plan Verification Plan}
 
 \wss{Create a checklists?}
 
+This document must be verified to make sure it is robust and reliable. The plan must have extensive coverage to ensure that all requirements are covered. The two main methods we will be using are the following:
+
+\begin{itemize}
+  \item \textbf{Mutation Testing}: This is essentially making small changes to the code to see if the the test cases can detect the changes. This ensures the test cases are robust.
```
Comment (gkaur034) [inline docs/VnVPlan/VnVPlan.tex:157]: 'the' is written twice

File: docs/VnVPlan/VnVPlan.tex [hunk 2]
```diff
@@ -147,6 +151,16 @@ \subsection{Verification and Validation Plan Verification Plan}
...
+  
+  \item \textbf{VnV Reviews}: With our Supervisor and Peers we will hold reviews this will help catch any potential gaps. This will help ensure completeness, accuracy, and feasibility of the plan.
```
Comment (gkaur034) [inline docs/VnVPlan/VnVPlan.tex:159]: Please rephrase the first line

File: docs/VnVPlan/VnVPlan.tex [hunk 3]
```diff
@@ -160,6 +174,16 @@ \subsection{Implementation Verification Plan}
 walkthrough.  There is also a possibility of using the final presentation (in
 CAS741) for a partial usability survey.}
 
+The implementation verification plan is really a lot of static review this is include code walkthroughs and code inspections in conjunction with static analyzers. The plan will be as follows:
+
+\begin{enumerate}
+  \item Static analysis will will be present at the CI/CD pipeline level essentially whenever code is committed or a pull request is created. This will help catch code errors or style violations.
```
Comment (gkaur034) [inline docs/VnVPlan/VnVPlan.tex:180]: 'will' is written twice

File: docs/VnVPlan/VnVPlan.tex [hunk 4]
```diff
@@ -160,6 +174,16 @@ \subsection{Implementation Verification Plan}
...
+  \item Next, the team will do a high level code walkthrough to catch any high level logic errors.
+  \item Finally, in any critical sections or features we will do a thorough code inspection where will go through a detailed checklist to ensure the implementation is verified.
```
Comment (gkaur034) [inline docs/VnVPlan/VnVPlan.tex:182]: change 'where will go' to 'where we will go'

File: docs/VnVPlan/VnVPlan.tex [hunk 5]
```diff
@@ -197,6 +223,8 @@ \subsection{Software Validation Plan}
 
 \wss{This section might reference back to the SRS verification section.}
 
+Validation of the project is critical and our supervisor will be vital for this process. Our supervisor will make sure that our Software solves the problem and it solves the user groups needs. Our supervisor is in an excellent position to validate the software as not only are they a domain expert they also fit the potential user group.
```
Comment (gkaur034) [inline docs/VnVPlan/VnVPlan.tex:226]: rephrase second line to be 'Our supervisor will make sure that our software solves the problem and the user group needs.' Add a comma after 'expert' in next line.

**Review by gkaur034 — CHANGES_REQUESTED** [review]
Looks good, but please fix according to my comments.

**Comment by PKALXI** [comment]
Please review @Inreet-Kaur  @gkaur034

**Pull request merged**